content/working-with-azure-key-vault-in-csharp/index.md

```diff
@@ -0,0 +1,455 @@
+
+    3.  Specify the Vault Uri in AppSettings
+
+    4.  Access Secrets from AzureKeyVault
+
+    5.  Retrieve Secret Value from AzureKeyVault
+
+    6.  Register the Dependencies in the ConfigureServices method
+
+5.  Deploy the Application to Azure
+
+6.  Execute the Application
+
+## **Prerequisites**
+
+To execute the code examples provided in this article you should have each of the following:
+
+1. Visual Studio 2019 (preferred)
+
+2. Azure Account
+
+
+If you don't have an Azure account, you can create one for free here:
+<https://portal.azure.com>
+
+## **What is Azure Key Vault?**
+
+Azure Key Vault is a cloud service that enables you to store your application's secrets in a safe and secure place. You can keep the keys, passwords, certificates, and other secrets safely and securely. Azure Key Vault is a service that lets you store secrets and other sensitive configuration data for a particular application. You can take advantage of Azure Key Vault to set parameters shared among multiple applications, including applications running in App Service.
+
+It enables you to isolate the sensitive and non-sensitive data in your application. For example, you can use application settings to store default parameters or key-value pairs containing some default settings used by the
+application. On the contrary, you can use Azure Key Vault to store API keys, secret keys, database connection strings, or Client IDs used in your application.
+
+## **Understanding Managed Identity**
+
+Managed identity is a concept that eliminates the need of having to store credentials once an application has been deployed to the cloud. By using managed identity, we can securely access a variety of Azure services without having to store any credentials like as connection strings or passwords. Managed identity may be used to connect to Key Vault from an Azure Function App or an Azure Web App, as well as to connect to Azure Blob Storage from an Azure Web App.
+
+Managed identities are of the following two types:
+
+-   System-assigned - this is activated directly on the Azure cloud service. For example, you can create a managed identity for your web app that will be automatically deleted once the web app instance is deleted. 
+    
+-   User-assigned - this is a stand-alone Azure resource that is created independent of the actual service instance. You can assign this identifier to one or more Azure resources simultaneously. As an example, you can set this identity to both an Azure Web App and an Azure Function at the same point in time.
+
+## **Create a Web App in the Azure Portal**
+
+We will now create an Azure Web App instance with the permissions to access Azure Key Vault. Adhere to the steps given below that would guide you to create a new Azure Web App instance while you’re within the Azure portal:
+
+1.  In the Azure Portal, click on "Create a resource" from the Home screen
+
+2.  Select Web App from the list
+
+![](media/881710611150de6c542f6656de3b727e.png)
```

Comment:
Add alt text in all the images in the blog [alt text](image path)

Reply:
Done, thanks!